### PR TITLE
Add disabling logic

### PR DESCRIPTION
**Pull request by DTMad**

### CODE REVIEW

File: controllers/namespace/namespace_controller.go [hunk 1]
```diff
@@ -12,7 +12,8 @@ import (
 	"github.com/Dynatrace/dynatrace-operator/webhook"
 	"github.com/go-logr/logr"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (lrgar) [inline controllers/namespace/namespace_controller.go:16]:
```suggestion
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```

File: controllers/namespace/namespace_controller.go [hunk 2]
```diff
@@ -85,16 +86,22 @@ func (r *ReconcileNamespaces) Reconcile(ctx context.Context, request reconcile.R
...
 
+	tokenName := utils.GetTokensName(&apm)
+	if !apm.Spec.CodeModules.Enabled {
+		r.ensureSecretDeleted(tokenName, targetNS)
+		return reconcile.Result{RequeueAfter: 5 * time.Minute}, nil
+	}
+
+	var ims dynatracev1alpha1.DynaKubeList
+	if err := r.client.List(ctx, &ims, client.InNamespace(r.namespace)); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to query OneAgentIMs: %w", err)
```
Comment (lrgar) [inline controllers/namespace/namespace_controller.go:102]:
```suggestion
		return reconcile.Result{}, fmt.Errorf("failed to query OneAgentIMs: %w", err)
```

Please also update `failed to query OneAgentAPM` above.

File: controllers/oneagent/oneagent_controller.go [hunk 3]
```diff
@@ -125,79 +128,89 @@ func (r *ReconcileOneAgent) Reconcile(ctx context.Context, rec *utils.Reconcilia
...
+	}
+	return dsDesired, nil
+}
+
+func (r *ReconcileOneAgent) getPods(ctx context.Context, instance *dynatracev1alpha1.DynaKube, feature string) ([]corev1.Pod, []client.ListOption, error) {
 	podList := &corev1.PodList{}
 	listOps := []client.ListOption{
 		client.InNamespace((*instance).GetNamespace()),
-		client.MatchingLabels(buildLabels((*instance).GetName())),
+		client.MatchingLabels(buildLabels((*instance).GetName(), feature)),
```
Comment (lrgar) [inline controllers/oneagent/oneagent_controller.go:213]: You don't need the dereference here,

```suggestion
		client.MatchingLabels(buildLabels(instance.Name, feature)),
```

Comment (DTMad) [reply controllers/oneagent/oneagent_controller.go]: Done

File: controllers/namespace/namespace_controller.go (hunk 2 above)
Comment (DTMad) [reply controllers/namespace/namespace_controller.go]: Done

File: controllers/namespace/namespace_controller.go (hunk 1 above)
Comment (DTMad) [reply controllers/namespace/namespace_controller.go]: Done

**Review by lrgar — APPROVED** [review]
Error messages should be different so that we know which line is the problem, but good otherwise.

**Review by lrgar — APPROVED** [review]

**Pull request merged**